modules/services/swayidle.nix

```diff
@@ -108,8 +108,9 @@ in {
 
       Service = {
         Type = "simple";
-        ExecStart =
-          "${cfg.package}/bin/swayidle -w ${concatStringsSep " " args}";
+        ExecStart = "${pkgs.writeShellScript "swayidle" ''
+          ${cfg.package}/bin/swayidle -w ${concatStringsSep " " args}
+        ''}";
```

Comment:
I don't think this fixes #2811.

``` console
$ env PATH= $(which sh) -c 'echo $PATH; sh -c "echo allo"'
/run/current-system/sw/bin/sh: line 1: sh: No such file or directory

$ env PATH= $(nix-build --no-out-link -E '(import <nixpkgs> { }).writeShellScript "test" "sh -c '"'echo hello'"'"')
/nix/store/lp5b7v1w2v6adf8rrmj3sxq84g0482a7-test: line 2: sh: No such file or directory
```

The default PATH environment variable in systemd services only contains the directory with systemd binaries.

``` console
$ systemctl --user show-environment | grep '\bPATH\b'
PATH=/nix/store/y314jq17kifh87h2snsma33ll494qnz0-systemd-249.7/bin/
```

One possible fix is to set `Service.Environment = [ "${dirOf pkgs.runtimeShell}" ]`.
This directory contains an `sh` binary.